stan/math/prim/fun/eigenvalues.hpp

```diff
@@ -7,14 +7,28 @@
 namespace stan {
 namespace math {
 
-template <typename EigMat, require_eigen_matrix_dynamic_t<EigMat>* = nullptr>
+template <typename EigMat, require_eigen_matrix_dynamic_t<EigMat>* = nullptr,
+          require_not_vt_complex<EigMat>* = nullptr>
 inline auto eigenvalues(const EigMat& m) {
```

Comment:
Is this return type `auto` because `solver.eigenvalues()` returns back a template expression?

I looked at the doc and it looks like it's not (but would believe you if you told me the doc wasn't correct). Would it be possible to change this to specify the (templated) return type for clarity?

https://eigen.tuxfamily.org/dox/classEigen_1_1EigenSolver.html#a0f507ad7ab14797882f474ca8f2773e7

Reply:
It's mostly for convenience, since the function accepts a template writing down the return type requires getting the scalar type out of the template and is verbose. I think it is, in full,

`Eigen::Matrix<complex_return_t<value_type_t<EigMat>, 1, -1>`

Reply:
For readability, I think it's worthwhile to include that type for both here and in `eigenvalues(...)`. Would you mind updating that?
(Yes, it's easier to write. But it makes it a lot harder to read later on.)

Reply:
Will do. Also, just noting for myself, I should change these template parameters to something like EigCplxMat to make it more clear as well

Reply:
👍